### PR TITLE
Update dash extension

### PR DESCRIPTION
**Pull request by JasonBoy**
## Description

Basicly Add 2 new commands to quickly search in specific docset: 
- `Create Docset Quicklink` command to create a deeplink to quickly jump into single docset search
- `Find in Specific Dash Docset` command, can use it directly, also it is used by the command above internally
- Add an readme, and some screenshot
- update raycast api to latest

## Screencast

https://github.com/raycast/extensions/assets/2911620/b0761e09-a699-4650-b1c0-d4f0018a63dd

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by JasonBoy** [comment]
Improvement for https://github.com/raycast/extensions/issues/2935

File: extensions/dash/package.json [hunk 1]
```diff
@@ -6,28 +6,50 @@
...
+      "description": "Create a quicklink for a specific docset in Dash.",
       "mode": "view",
       "arguments": [
         {
-          "name": "searchstring",
-          "placeholder": "Query",
-          "type": "text",
-          "required": false
+          "name": "docset",
+          "placeholder": "Docset keyword",
```
Comment (pernielsentikaer) [inline extensions/dash/package.json:45]:
```suggestion
          "placeholder": "Docset keyword",
          "required": true,
```

File: extensions/dash/src/create-docset-link.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+export default function Command(props: { arguments: DashArguments }) {
+  const { docset = "" } = props.arguments;
+  if (!docset) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Docset keyword is required",
+    });
+    return <Detail markdown={`Can not create Quicklink, docset keyword is required!`}></Detail>;
+  }
```
Comment (pernielsentikaer) [inline extensions/dash/src/create-docset-link.tsx:13]:
```suggestion
  const { docset } = props.arguments;
```

File: extensions/dash/package.json [hunk 3]
```diff
@@ -6,28 +6,50 @@
...
+          "type": "text"
+        }
+      ]
+    },
+    {
+      "name": "create-docset-link",
+      "title": "Create Docset Quicklink",
+      "subtitle": "Docset keyword in Dash",
+      "description": "Create a quicklink for a specific docset in Dash.",
       "mode": "view",
```
Comment (pernielsentikaer) [inline extensions/dash/package.json:41]: This should be a [no-view](https://developers.raycast.com/information/lifecycle#launch) command instead, that would save the end user a few clicks 🙂

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
@RSO do you mind checking this too 🙂

File: extensions/dash/package.json (hunk 3 above)
Comment (JasonBoy) [reply extensions/dash/package.json:41]: @pernielsentikaer  Thanks for the review, for this one, I try to make it no-view, but in no view, when I `launchCommand` for `create-quicklink` in the extension, the link I passed to `arguments` won't be autofilled into the `Link` field of the quicklink form opened, user need to paste that manually(call `Clipboard.copy` before), So I change it to view command with an `Action`(which can autofill the form item), do you know how to achieve that with no-view command, I would happy to change that 😃

File: extensions/dash/src/create-docset-link.tsx (hunk 2 above)
Comment (JasonBoy) [reply extensions/dash/src/create-docset-link.tsx:13]: If it's still a view command, no need to show a simple message? I define it as `required` in package.json, but I can still open the command by press enter while I enter nothing in the command query, is it possible to reject the open command (like in the `no-view` command to just show a toast in the bottom left corner without opening the command) when the query is required?

File: extensions/dash/package.json (hunk 3 above)
Comment (pernielsentikaer) [reply extensions/dash/package.json:41]: You could do it like this:
```
import { DashArguments } from "./types";
import { launchCommand, LaunchType } from "@raycast/api";
import { createDeeplinkForDocset } from "./utils";

export default async function Command(props: { arguments: DashArguments }) {
  const { docset = "" } = props.arguments;
  const deeplink = createDeeplinkForDocset(docset);

  await launchCommand({
    ownerOrAuthorName: "raycast",
    extensionName: "raycast",
    name: "create-quicklink",
    type: LaunchType.UserInitiated,
    fallbackText: deeplink,
  });
}
```

Comment (JasonBoy) [reply extensions/dash/package.json]: thanks for the help

**Comment by RSO** [comment]
Looks good to me! Thanks for the contribution 😄

**Pull request merged**